src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/ViewModels/ValidationProblemViewModelTestExtensions.cs

```diff
@@ -0,0 +1,244 @@
+        IEnumerable<string> allowedProperties)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.LocationAllowedProperty.Code
+        );
+
+        var errorDetail = error.GetDetail<LocationStringValidators.AllowedPropertyErrorDetail>();
+
+        Assert.Equal(property, errorDetail.Value.Property);
+        Assert.Equal(allowedProperties, errorDetail.AllowedProperties);
+
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasLocationMaxValueLengthError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        string value,
+        int maxValueLength)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.LocationMaxValueLength.Code
+        );
+
+        var errorDetail = error.GetDetail<LocationStringValidators.MaxValueLengthErrorDetail>();
+
+        Assert.Equal(value, errorDetail.Value.Value);
+        Assert.Equal(maxValueLength, errorDetail.MaxValueLength);
+
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasSortFormatError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        string? value)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.SortFormat.Code
+        );
+
+        var errorDetail = error.GetDetail<FormatErrorDetail>();
+
+        Assert.Equal(value, errorDetail.Value);
+        Assert.Equal("{field}|{order}", errorDetail.ExpectedFormat);
+
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasSortMaxFieldLengthError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        string field,
+        int maxFieldLength = 40)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.SortMaxFieldLength.Code
+        );
+
+        var errorDetail = error.GetDetail<SortStringValidators.MaxFieldLengthErrorDetail>();
+
+        Assert.Equal(field, errorDetail.Value.Field);
+        Assert.Equal(maxFieldLength, errorDetail.MaxFieldLength);
+
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasSortDirectionError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        string direction)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.SortDirection.Code
+        );
+
+        var errorDetail = error.GetDetail<SortStringValidators.DirectionErrorDetail>();
+
+        Assert.Equal(direction, errorDetail.Value.Direction);
+        Assert.Equal(["Asc", "Desc"], errorDetail.AllowedDirections);
```

Comment:
Same here, I'd move this check to `SortStringValidatorsTests` and only check the `direction` here